### PR TITLE
Contact form

### PR DESCRIPTION
**Pull request by barbarah**

### CODE REVIEW

File: apps/dataset-browser/src/messages/en/messages.json [hunk 1]
```diff
@@ -128,5 +128,17 @@
...
     "message": "Something went wrong!"
+  },
+  "ContactForm": {
+    "labelEmailAddress": "Email",
+    "labelSubject": "Subject",
+    "labelMessage": "Message",
+    "sendEmailButton": "Send message",
+    "emailIsSended": "Your message has been sent successfully.",
+    "communityEditServerError": "There was a problem sending your message. Please try again later.",
+    "emailAddressInvalidString": "Email address is not valid",
```
[the remark this answers is not in the document]
Comment (barbarah) [reply apps/dataset-browser/src/messages/en/messages.json:139]: This is based on the code `t(camelCase(`emailAddress_${errors.emailAddress.type}`))}`

Where `errors.emailAddress.type` will be `invalid_string`

File: apps/dataset-browser/src/messages/en/messages.json [hunk 2]
```diff
@@ -128,5 +128,17 @@
...
+    "subjectTooSmall": "Subject is required",
+    "subjectTooBig": "Subject has a maximum of 250 characters",
```
[the remark this answers is not in the document]
Comment (barbarah) [reply apps/dataset-browser/src/messages/en/messages.json:141]: This is based on the code ``t(camelCase(`subject_${errors.subject.type}`))``

Where `errors.subject.type` will be `too_big`. The type name is determined by `zodResolver`.

File: apps/dataset-browser/src/messages/en/messages.json [hunk 3]
```diff
@@ -128,5 +128,17 @@
...
   "ErrorPage": {
     "message": "Something went wrong!"
+  },
+  "ContactForm": {
+    "labelEmailAddress": "Email",
+    "labelSubject": "Subject",
+    "labelMessage": "Message",
+    "sendEmailButton": "Send message",
+    "emailIsSended": "Your message has been sent successfully.",
+    "communityEditServerError": "There was a problem sending your message. Please try again later.",
```
[the remark this answers is not in the document]
Comment (barbarah) [reply apps/dataset-browser/src/messages/en/messages.json:138]: Whops, I used the community edit form as a baseline for this form, but I forgot to rename the error key.

File: apps/dataset-browser/src/messages/en/messages.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdevalk) [reply apps/dataset-browser/src/messages/en/messages.json:139]: Thanks - I missed this!

This implementation leads to a tight coupling between our locale keys and Zod's issue codes, though, and additional logic to bridge these (e.g. by using camelcasing). I'm fine with this if there is no other way to do is. To be sure: is there an alternative? For example: can we make use of an [error map](https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#customizing-errors-with-zoderrormap) instead?

[the remark this answers is not in the document]
Comment (barbarah) [reply apps/dataset-browser/src/messages/en/messages.json]: I have a tight coupling between our locale keys and Zod's issue codes by adding the translation key to the schema.

**Review by sdevalk — APPROVED** [review]

File: apps/dataset-browser/src/messages/en/messages.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdevalk) [reply apps/dataset-browser/src/messages/en/messages.json]: Nice solution! 🎉

**Pull request merged**